### ISSUE TITLE
Provide keyboard shortcuts for navigating through the list of files

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-code-manager#199]
~~Depends on #185.~~
~~Depends on #459.~~
~~Depends on #469.~~
~~Depends on #471.~~

This came up in our walkthrough of the current reviewer tools. They include the following shortcuts when viewing the contents of a version:

<img width="187" alt="screenshot 2019-02-13 11 39 34" src="https://user-images.githubusercontent.com/142755/52727833-12da5d80-2f84-11e9-9ba9-27bc9849595c.png">

I'm not sure if we need all of these, so there should be some more discussion about what is actually required. I believe we also decided that this isn't part of the MVP.

**Comment by bobsilverberg** [comment]
I understand that this was discussed at the Show & Tell demo. 

@wagnerand Can you comment on which of these we need? "Up file" and "Down file" seem to make sense, but I'm not sure that "Previous note" and "Next note" make sense anymore. I assume "Expand all" means to expand the whole tree? Do we still want/need "Hide or unhide tree"? What about "Wrap or unwrap text", does that still make sense with the current implementation?

**Comment by wagnerand** [comment]
I would keep the following:

* Up file
* Down file
* Next diff (for diff-view)
* Previous diff (for diff-view)
* Hide/Unhide tree

If _Next/Previous note_ is a linter/validation message, it would be great if we could keep them as well. If that's something else, let me know.

**Comment by bobsilverberg** [comment]
Thanks @wagnerand.

> If Next/Previous note is a linter/validation message, it would be great if we could keep them as well. If that's something else, let me know.

I don't know what "Next/Previous note" are. That's why I was asking you. 

I tried out `k - Up file`, `j - Down file`, `p - Previous note` and `n - Next note` on an existing review (https://reviewers.addons.allizom.org/en-US/firefox/files/browse/327334), and only one of them seem to behave as expected. `j - Down file` and `n - Next note` just seem to jump the selection to the next file in the list. `k - Up file` and `p - Previous note` don't seem to do anything logical. Wherever I am in the file tree, each of those jumps the selection to `Chart.min.js`, which seems odd. Trying `k - Up file` and `p - Previous note` on a different extension (https://reviewers.addons.allizom.org/en-US/firefox/files/browse/709922), it seems like maybe they just jump the selection to the bottom-most selectable file.

Is anyone in a position to explain what each of these is actually supposed to do?

Also note that you didn't include "Expand all" in your list, but I actually just finished implementing that, so I think we might as well include it. We're going to make it buttons above the File Tree for now, so maybe we don't bother including keyboard shortcuts for it, but it won't really cost us anything to also have keyboard shortcuts for it.

**Comment by wagnerand** [comment]
Sorry for that. I probably should have added more context.

Lots of the current commands are broken right now, especially the "previous" ones. Also, I believe they automatically skip non-code files (?) (based on the file type/extension), yay magic!

I checked the implementation for _Next/Previous note_. They are indeed linter messages, but they are only shown, when we are **not** in diff-mode. In diff-mode, we swap them out for _Next/Previous diff_, ugh. I think in the new implementation, we could show those commands always, if that makes things easier.

I can explain what they are *supposed* to do though:

* Up file: Go one file up.
* Down file: Go one file down.
* Next diff: Go to next diff block in the current file. If at last diff block in the current file, go to first diff block in the next file.
* Previous diff: Analogue to _Next diff_.
* Next note: Go to next linter annotation in the current file. For consistency we, can jump to first annotation in the next file (or just do nothing).
* Previous note: Analogue to _Previous note_.
* Hide/Unhide tree: Hides/Unhides the file tree on the left to provide more screen space for the current file.

The following applies to _Next diff_, _Previous diff_, _Next note_ and _Previous note_ equally and respectively , but I will only refer to one for the sake of readability:
We should probably figure out what to do when we already are at the file boundary, ie last diff  in a file. Going to the first diff in the next file would be great, as it would allow mouseless-navigation basically through all files. Please let me know if that is not feasible or lots of effort and we'll try to come up with another approach.

> Also note that you didn't include "Expand all" in your list, but I actually just finished implementing that, so I think we might as well include it.

Great. I didn't think it's super important which is why I didn't mention it. But even better if it's already done. :)

**Comment by bobsilverberg** [comment]
This has been fully implemented. Hooray!

**Comment by bobsilverberg** [comment]
Actually, I need to make sure we have issues for the shortcuts that have not been implemented yet. I will do that.
